### PR TITLE
fix: PFD missing function

### LINKED ISSUE #3719: [BUG] Pilot-side PFD was frozen for a long time
**Mod Version**

A32NX Experimental Custom FBW

**Describe the bug**

I had something weird happening on my flight last night with the Custom FBW version. It's mainly on the Pilot side PFD. For some reason it was frozen. The airspeed wasn't changing the was it's suppose to and the altitude stays. When I look at the co-pilot side it's makes much more sense because it's displaying the correct data and it's still moving. So I thought of turning off the Pilot PFD, and I waited.. and waited.... and waited... and the screen never goes past the self-test for a long while. So I had to make a decision and switch the pilot flying to copilot, and switches my TCA Quadrant and sidestick the other way for a right-hand flying. After a few more minutes, I look at the Pilot PFD, it's still on self-test.. After a while, I look at it again and it's now working. So I switched back my camera and TCA Quadrant and sidestick back to Pilot flying and I end up landed just fine. My issue is what happened in the first place? Attached is the video evidence.

**To Reproduce**

1. Fly as usual and starting approach.
2. PFD Suddenly frozen, the airspeed and altitude doesn't change to reflect the actual behaviour.
3. Check with Copilot-side PFD, and it's showing a different data that reflects the actual behaviour and it's not frozen.
4. Turn down the brightness knob to reset the Pilot-side PFD. And turn it back on to go to self-test mode.
5. The Self-test mode went far over the 40 second waiting time.
6. The PFD is unusable until short final.

**Expected behavior**

PFD should keep refreshing with real time data of the aircraft behaviour at all times.

**Actual behavior**

Pilot-side PFD is frozen. It doesn't move the airspeed and altitude changes. When I turn off the display, and turn it on. It stays frozen in Self-Test mode for a long while.

**References**

https://youtu.be/IymFrB-xqc0?t=9083

**Additional context**

Was this working before/when did the issue start occurring?
- This is the first time I experience this bug. And I haven't flown enough to see if this is a persistent or consistent bug or if it's just a single incident.

Is this a problem in the vanilla unmodded game?
- I haven't checked.

Discord username (if different from GitHub): Neo 4316

### PR DESCRIPTION
**Pull request by Kimbyeoungjang**
Fixes #3719

## Summary of Changes

I really sorry about that.

MY PR BROKE PFD..... Add missing function on that. 

## Additional context

Discord username (if different from GitHub):

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Review by Benjozork — APPROVED** [review]
LGTM

**Comment by marcman86** [comment]
Category: QA Tester
Name: marcman86#4907
Date of testing: 28.2.2021
Version of the sim: 1.13.16.0
PR Tested: #3748
Tier of testing: 2
Changes to observe: PFD fix
Testing technique: Tryout

Steps made to test the Pull Request:

1.) Spawned Runway at EDDP for a round back to EDDP
2.) Get to CRZ alt. 5000ft 
3.) After Star decent to 4000ft
4.) Capture ILS and land

Its unclear in which moment the freeze appears. So hard to reproduce it in one try. But in this small round it happens almost every time before the fix.

Issues: -

Overall Rating: Good

Conclusions of the testing: Seems to fix it.

Hope this helps :-)

**Comment by BjoernHund** [comment]
Category: Tester
Name: Björn#7582
Date of testing: 28.02.2021
Version of the sim: 1.13.16.0
PR Tested: #3748
Tier of testing: 2
Changes to observe: PFD fix
Testing technique: Tryout

Steps made to test the Pull Request:

1. Full flight from EDDN to EDDS
2. Climb to CRZ alt. FL160
3. LBU descend to 5000 
3.) STAR descend to 4000
4.) Capture ILS and land

Issues: -

Overall Rating: Good - Works like a charm

**Comment by sidnov** [comment]
@Kimbyeoungjang Glad that the bug was found :D Testing now

**Review by davidwalschots — COMMENTED** [review]
I'm confused when I compare this to the previous change. Are you sure not using the `getAutoPilotSelectedAltitudeLockValue` is intended? If so, then shouldn't the comment be changed or removed?

image

**Comment by Kimbyeoungjang** [comment]
> I'm confused when I compare this to the previous change. Are you sure not using the `getAutoPilotSelectedAltitudeLockValue` is intended? If so, then shouldn't the comment be changed or removed?
> 
> image

yes because the getAutopilotSelectedAltitudeLockValue only change altitude when press or pull knob.

IRLA320 pilot report it is not. Also can do FCU change altitude via turn knob. So need to use FCU displayed value.

hhe this.hudAPAltitude is FCU displayed altitude. so that is working as expect.

**Comment by davidwalschots** [comment]
Okay. Then my comment still stands: shouldn't part of or the whole comment be removed?
> // Use the constraint altitude if provided otherwise use **selected altitude lock value**

**Comment by Kimbyeoungjang** [comment]
> Okay. Then my comment still stands: shouldn't part of or the whole comment be removed?
> 
> > // Use the constraint altitude if provided otherwise use **selected altitude lock value**

Oh I see :)

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/PFD/AltimeterIndicator.js [hunk 1]
```diff
@@ -487,7 +487,7 @@ class Jet_PFD_AltimeterIndicator extends HTMLElement {
 
         // Use the constraint altitude if provided otherwise use selected altitude lock value
```
Comment (Kimbyeoungjang) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/PFD/AltimeterIndicator.js:488]:
```suggestion

```

**Review by Kimbyeoungjang — COMMENTED** [review]
remove remark

**Comment by marcman86** [comment]
Category: QA Tester
Name: marcman86#4907
Date of testing: 28.2.2021
Version of the sim: 1.13.16.0
PR Tested: #3748
Tier of testing: 2
Changes to observe: PFD fix
Testing technique: Tryout

Steps made to test the Pull Request:

1.) Spawned Runway at EDDT for a round back to EDDT
2.) Get to CRZ alt. 15000ft
3.) After Star decent to 3000ft
4.) Capture ILS and land

Issues: -

Overall Rating: Good

Conclusions of the testing: Works again

Hope this helps :-)

**Pull request merged**